src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,276 @@
+                _path);
+        }
+
+        /// <summary>
+        /// Calls the necessary Spark functions to create org.apache.spark.broadcast.Broadcast
+        /// object for Spark versions 2.3.2 and above, and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="javaSparkContext">Java Spark context object</param>
+        /// <param name="sc">SparkContext object</param>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_2_Above(
+            JvmObjectReference javaSparkContext,
+            SparkContext sc,
+            string tempDir,
+            object value)
+        {
+            var pythonBroadcast = (JvmObjectReference)javaSparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.api.python.PythonRDD",
+                "setupBroadcast",
+                _path);
+            
+            SparkConf sparkConf = sc.GetConf();
+            bool encryptionEnabled = bool.Parse(
+                sparkConf.Get("spark.io.encryption.enabled", "false"));
+
+            if (encryptionEnabled)
+            {
+                throw new NotImplementedException("Broadcast encryption is not supported yet.");
+            }
+            else
+            {
+                WriteBroadcastValueToFile(tempDir, value);
+            }
+
+            return (JvmObjectReference)javaSparkContext.Jvm.CallNonStaticJavaMethod(
+                javaSparkContext,
+                "broadcast",
+                pythonBroadcast);
+        }
+    }
+
+    /// <summary>
+    /// Global registry to store the object value of all active broadcast variables from
+    /// the workers. This registry is only used on the worker side when Broadcast.Value() is called
+    /// through a UDF.
+    /// </summary>
+    internal static class BroadcastRegistry
+    {
+        private static readonly ConcurrentDictionary<long, object> s_registry =
+            new ConcurrentDictionary<long, object>();
+
+        /// <summary>
+        /// Function to add the value of the broadcast variable to s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to add</param>
+        /// <param name="value">Value of the Broadcast variable</param>
+        internal static void Add(long bid, object value) => s_registry.TryAdd(bid, value);
+
+        /// <summary>
+        /// Function to remove the Broadcast variable from s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to remove</param>
+        internal static void Remove(long bid)
+        {
+            bool result = s_registry.TryRemove(bid, out _);
+            if (!result)
+            {
+                throw new Exception($"Trying to remove an already removed broadcast with id {bid}.");
```

Comment:
`already removed` is a bit misleading. Can you reword it so that it's a removing something that doesn't exist?

Reply:
How about `Trying to remove a Broadcast {bid} that does not exist`